services/skus/controllers.go

```diff
@@ -1188,12 +1189,53 @@ func handleWebhookAppStoreH(w http.ResponseWriter, r *http.Request, svc *Service
 }
 
 // handleRadomWebhook handles Radom checkout session webhooks.
-func handleRadomWebhook(_ *Service) handlers.AppHandler {
+func handleRadomWebhook(s *Service) handlers.AppHandler {
 	return func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
-		return handlers.RenderContent(context.Background(), struct{}{}, w, http.StatusNotImplemented)
+		return handleRadomWebhookH(w, r, s)
 	}
 }
 
+func handleRadomWebhookH(w http.ResponseWriter, r *http.Request, svc *Service) *handlers.AppError {
+	ctx := r.Context()
+
+	l := logging.Logger(ctx, "skus").With().Str("func", "handleRadomWebhookH").Logger()
+
+	if err := svc.radomAuth.Authenticate(ctx, r.Header.Get("radom-verification-key")); err != nil {
+		l.Error().Err(err).Msg("invalid request")
+
+		return handlers.WrapError(err, "invalid request", http.StatusUnauthorized)
+	}
+
+	b, err := io.ReadAll(io.LimitReader(r.Body, reqBodyLimit10MB))
+	if err != nil {
+		l.Err(err).Msg("failed to read payload")
+
+		return handlers.RenderContent(ctx, struct{}{}, w, http.StatusOK)
+	}
+
+	event, err := radom.ParseEvent(b)
+	if err != nil {
+		l.Err(err).Msg("failed to parse radom event")
+
+		return handlers.WrapError(err, "failed to parse radom event", http.StatusBadRequest)
+	}
+
+	if err := svc.processRadomEvent(ctx, event); err != nil {
+		l.Err(err).Msg("failed to process radom event")
+
+		return handlers.WrapError(model.ErrSomethingWentWrong, "something went wrong", http.StatusInternalServerError)
+	}
```

Comment:
@clD11 This is an open important question.

Reply:
For the mobile handlers i.e. `handleWebhookPlayStoreH` and `handleWebhookAppStoreH` things are quite different with regards to the order i.e. in many cases we will not have an order due to not linking however this should never be the case for Radom.

Also looking at the other handlers its not clear how `model.ErrNoRowsChangedOrderPayHistory` can ever be thrown. Unless I am missing something the [current code](https://github.com/brave-intl/bat-go/blob/d35e02760946bbce41cba0786de2abc9074d58a9/services/skus/storage/repository/order_history.go#L30) will either insert or throw a [unique constraint violation](https://github.com/brave-intl/bat-go/blob/d35e02760946bbce41cba0786de2abc9074d58a9/migrations/0036_order_time.up.sql#L9) so `model.ErrNoRowsChangedOrderPayHistory` will never been returned. Whats puzzling is how this is working for the other handlers with regards to duplicates and I see we are setting the [paid at timestamp](https://github.com/brave-intl/bat-go/blob/d35e02760946bbce41cba0786de2abc9074d58a9/services/skus/service.go#L1640) to the time we process the message rather than the actual paid at time hence it always inserts a new row even for duplicates, this will not be the case for Radom as it uses the actual paid at time which will be consistent.

Lets take this one 'offline' and discuss how we want to handle the events as there are a couple of things we might want to address here.

Reply:
Yes, let's discuss it offline.

The main point I've made in/by the comment is that responding with a `500` in all possible errors here is not necessarily good enough. Maybe some granularity, if possible, is good.